### PR TITLE
Migrate Roadworks Layer

### LINKED ISSUE #385: Migrate roadworks map layer from master to next branch
https://herrenberg.stadtnavi.de/ (master branch) has a feature were roadworks are shown on the map. To enable it, use the map layer selector at the bottom left.

Screenshot from 2021-05-05 16-16-17

Please migrate this map layer to the `next` branch.

This feature has been reworked a bunch of times so there isn't a single PR that I can point you towards but looking at the history of `Roadworks.js` should give you a couple of pointers: https://github.com/stadtnavi/digitransit-ui/commits/master/app/component/map/tile-layer/Roadworks.js

Other relevant files from the master branch are:

- https://github.com/stadtnavi/digitransit-ui/blob/master/app/component/map/tile-layer/Roadworks.js
- https://github.com/stadtnavi/digitransit-ui/blob/master/app/component/map/popups/RoadworksPopup.js
- https://github.com/stadtnavi/digitransit-ui/blob/master/app/component/map/tile-layer/RoadworksRow.js

*Note*: All the other map layers simply draw an icon on the map however this one has some custom code to use the canvas API to draw a line. If you have a question about this, please contact @leonardehrenfried.

What is a row?

The different row files refer to a the little selection lines that show up when you click on an area on the map which has more than one map feature:

Screenshot from 2021-05-05 16-24-07

### New features

- when a roadworks is in the future make the map feature (icon, polyline) sem-transparent
- use the the following icon if the closure is partial: https://commons.wikimedia.org/wiki/File:Zeichen_123_-_Arbeitsstelle,_StVO_2013.svg How a partial closure is defined is still a work in progress, so speak to @leonardehrenfried about the details.

### PR DESCRIPTION
**Pull request by andreashelms**
## Proposed Changes

  - Semi transparent Polyline if roadworks are in the future

image

  - Changed icon for partial closure

image

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform

closes #385

### CODE REVIEW

File: app/component/map/tile-layer/TileContainer.js [hunk 1]
```diff
@@ -231,22 +238,16 @@ class TileContainer {
...
           );
-        }
-        if (dist < 22 * this.scaleratio) {
-          return true;
-        }
-        return false;
+
+          return dist < 22 * this.scaleratio;
+        });
       });
```
Comment (andreashelms) [inline app/component/map/tile-layer/TileContainer.js:251]: @leonardehrenfried I really don't know how this works and how I can get things merged with master branch here. Can you please help me here or give a suggestion?

Comment (leonardehrenfried) [reply app/component/map/tile-layer/TileContainer.js:251]: The code before always assumed hat a feature is always a point however in this case we want to make the entire polyline clickable. That's what this code is about.

Comment (leonardehrenfried) [reply app/component/map/tile-layer/TileContainer.js:251]: Are you seeing an error?

**Comment by leonardehrenfried** [comment]
Great turnaround time, @andreashelms. I will help you over the finish line.

File: app/util/mapLayerUtils.js [hunk 2]
```diff
@@ -42,7 +42,10 @@ export const isFeatureLayerEnabled = (
   if (!Object.keys(mapLayers).includes(layerName)) {
     return false;
   }
-  const featureType = (feature.properties.type || '').toLocaleLowerCase();
+  const featureType =
+    layerName === 'roadworks'
+      ? null
+      : (feature.properties.type || '').toLocaleLowerCase();
```
Comment (andreashelms) [inline app/util/mapLayerUtils.js:48]: I implemented this temporarily to make the polyline click work. Can you please tell me where the map layer configuration is done and how I can add featureTypes?

Comment (leonardehrenfried) [reply app/util/mapLayerUtils.js:48]: i will take a look.

File: app/component/map/tile-layer/TileContainer.js (hunk 1 above)
Comment (andreashelms) [reply app/component/map/tile-layer/TileContainer.js:251]: Yes, I got that. But there are some changes here on next branch concerning `collision check for stops and citybike stations` and I am not sure how to get this together with polyline collision checks. But I will try.

Comment (leonardehrenfried) [reply app/component/map/tile-layer/TileContainer.js:251]: I see, i will take a look.

**Comment by leonardehrenfried** [comment]
Good work, thanks!

**Pull request merged**